### PR TITLE
Updates and migration prep

### PR DESCRIPTION
**Pull request by ryanblock**
This PR contains a variety of updates and preps for migrating off of ci.begin.com. Highlights:
- App now defaults to `nodejs20.x`, which is the current version running in Lambda, and that can be deployed after July's `nodejs16.x` deprecation
- Updates core dependencies; no longer relies on `aws-sdk` (deprecated), uses [`aws-lite`](https://aws-lite.org)
- No longer uses unmaintained `github-api` dependency, which had multiple subdependency security issues
- Tidies up all the `package[-lock].json` files all over the place that are no longer necessary in later versions of Architect
- Uses latest linter ljharb rules

This PR assumes moving from ci.begin.com to directly deploying to AWS via [Architect](https://arc.codes). If/when approved, we'll have to undertake some manual migration steps, but nothing too gnarly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
-  "name": "begin-app",
+  "name": "tc39-ci",
   "version": "0.0.7",
-  "description": "Begin basic Hello World! app",
+  "description": "Publishes TC39 drafts and updates to S3",
   "scripts": {
     "lint": "eslint . --ignore-pattern node_modules --fix",
-    "start": "cross-env NODE_ENV=testing npx sandbox",
-    "test": "echo 'tests skipped' || cross-env NODE_ENV=testing PORT=6666 tape test/*-test.js ./test/**/*-test.js | faucet",
+    "start": "cross-env npx sandbox",
```
Comment (ljharb) [inline package.json:7]:
```suggestion
    "start": "npx sandbox",
```
we don't need cross-env if we're not setting env vars, but why aren't we setting NODE_ENV anymore?

File: package.json [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
+    "test": "echo 'tests skipped' || cross-env PORT=6666 tape test/*-test.js ./test/**/*-test.js | faucet",
```
Comment (ljharb) [inline package.json:8]:
```suggestion
    "test": "echo 'tests skipped' || cross-env PORT=6666 tape 'test/**/*-test.js' | faucet",
```

File: src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js [hunk 3]
```diff
@@ -21,8 +21,8 @@ const [
 if (!slug || !branch) {
 	throw 'args required: slug, branch';
 }
-if (!process.env.GH_TOKEN) {
-	throw 'GH_TOKEN env var required';
+if (!process.env.GITHUB_TOKEN) {
```
Comment (ljharb) [inline src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js:24]: does github provide both of these by default?

File: src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js [hunk 4]
```diff
@@ -83,7 +83,7 @@ const authorsP = request(branchURL).then((json) => JSON.parse(json)).then((data)
 
 const teamURL = 'https://api.github.com/orgs/tc39/teams/delegates';
 
-function getMembers(teamID, page = 1) {
+const getMembers = (teamID, page = 1) => {
```
Comment (ljharb) [inline src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js:86]: why this change?

File: src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js [hunk 5]
```diff
@@ -92,26 +92,24 @@ function getMembers(teamID, page = 1) {
...
-			console.log(`${username} is a googler`);
-			return true;
-		},
-		() => {
-			console.log(`${username} is not a googler`);
-			return false;
-		},
-	);
-}
+const isGoogler = (username) => request(`https://api.github.com/orgs/googlers/members/${username}`).then(
```
Comment (ljharb) [inline src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js:103]: also why this change?

File: src/http/get-ipr-000user-000repo-pull-000prEtc/index.js [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
 const path = require('path');
 const { execSync } = require('child_process');
-const GitHub = require('github-api');
+const { Octokit } = require('@octokit/rest');
 
-const gh = new GitHub();
+const octokit = new Octokit({});
 
-function html({
+const html = ({
```
Comment (ljharb) [inline src/http/get-ipr-000user-000repo-pull-000prEtc/index.js:11]: and this one?

File: src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js (hunk 4 above)
Comment (ryanblock) [reply src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js:86]: All changes related to converting function expressions this way were to satisfy linter warnings I got from the current linting config (`"@ljharb/eslint-config": "^21.1.0`, `"eslint": "=8.8.0"`; I'm guessing perhaps it's this: https://github.com/ryanblock/tc39-ci/blob/f5c7d191712c114d2f987eb9047f6d588d0118f3/.eslintrc.js#L17

Would you like to see all those converted back? I don't personally care, I just assumed that's how you wanted it!

File: src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js (hunk 3 above)
Comment (ryanblock) [reply src/http/get-ipr-000user-000repo-pull-000prEtc/check-form.js:24]: All this is running in Lambda, so the token env var is whatever we want it to be. I observed the app was using both `GITHUB_TOKEN` + `GH_TOKEN`, so I figured we should standardize around one. It can be `GH_TOKEN` though, whatever works!

File: package.json (hunk 1 above)
Comment (ryanblock) [reply package.json:7]: In Arc 10 (2022) Arc removed all references to `NODE_ENV`; we found that too many other frameworks and tools were mutating it, so now it relies on its own `ARC_ENV` env var. Of course, everyone is still welcome to use `NODE_ENV` in their business logic, we just don't rely on that any longer.

File: package.json (hunk 2 above)
Comment (ryanblock) [reply package.json:8]: Per above, I think we may need cross env to properly glob there? Not sure but we don't have any tests here anyway, so no big deal.

File: src/http/get-ipr-000user-000repo-pull-000prEtc/index.js (hunk 6 above)
Comment (ryanblock) [reply src/http/get-ipr-000user-000repo-pull-000prEtc/index.js:11]: → https://github.com/ljharb/tc39-ci/pull/27#discussion_r1592709066

File: package.json (hunk 2 above)
Comment (ljharb) [reply package.json]: i don't think cross-env has any effect on globbing?

File: package.json (hunk 1 above)
Comment (ljharb) [reply package.json]: what's wrong with other things mutating it? isn't "whatever it's set to" meant to be the right env value?

**Review by ljharb — APPROVED** [review]
